src/batches/setBatchSegmentationDetectLesion.m

```diff
@@ -0,0 +1,37 @@
+% (C) Copyright 2021 CPP BIDS SPM-pipeline developers
+
+function matlabbatch = setBatchSegmentationDetectLesion(matlabbatch, BIDS, opt, subID)
+  %
+  % Creates a batch to segment the anatomical image for lesion detection
+  %
+  % USAGE::
+  %
+  %   matlabbatch = setBatchSegmentationDetectLesion(matlabbatch, BIDS, opt, subID)
+  %
+  % :param matlabbatch: list of SPM batches
+  % :type matlabbatch: structure
+  %
+  % :returns: - :matlabbatch: (structure)
+
+  printBatchName('Segmentation for lesion detection');
+
+% find anatomical file
+[anatImage, anatDataDir] = getAnatFilename(BIDS, subID, opt);
+matlabbatch{1}.spm.tools.ali.unified_segmentation.step1data = [anatImage, anatDataDir];
```

Comment:
OK the next thing would be to make sure that this function can now be used to extend a batch that already exists.

Here what it does is simply forcing the first element of the cell `matlabbatch` to be the step 1 of the ALI toolbox.

So we can to make sure this function just appends something at the end of that cell.

In our other batches we do that by using `end+1` for the first time set something to this cell. And then we just use `end`.

Something liked this.

```
matlabbatch{end + 1}.spm. ...
matlabbatch{end}.spm. ...
```

See here for a full example.

https://github.com/cpp-lln-lab/CPP_SPM/blob/c3511c754cb5bbe1de6b58a0189c91d57b3c0094/src/batches
/setBatchComputeVDM.m#L27

Another way to write those batch that I want to start using is the following: I find it easier to read.

```
unified_segmentation.step1prior = {lesionPriorMap};
unified_segmentation.step1niti = 2;                       % number of iterations
unified_segmentation.step1thr_prob = 0.333333333333333;   % threshold probability
unified_segmentation.step1thr_size = 0.8;                 % threshold size (in cm3) 
unified_segmentation.step1coregister = 1;                  % coregister in MNI space (yes: 1)
unified_segmentation.step1mask = {''};                     % specify cost function mask CFM (optional)
unified_segmentation.step1vox = 2;                         % Voxel sizes (in mm)
unified_segmentation.step1fwhm = [8 8 8];  

matlabbatch{end+1}.spm.tools.ali.unified_segmentation = unified_segmentation;
```

Will let you pick the method you want.